hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/run/strategy/MultipleSparkJobExecutionStrategy.java

```diff
@@ -91,13 +92,11 @@ public MultipleSparkJobExecutionStrategy(HoodieTable table, HoodieEngineContext
     // execute clustering for each group async and collect WriteStatus
     JavaSparkContext engineContext = HoodieSparkEngineContext.getSparkContext(getEngineContext());
     // execute clustering for each group async and collect WriteStatus
-    Stream<JavaRDD<WriteStatus>> writeStatusRDDStream = clusteringPlan.getInputGroups().stream()
+    Stream<JavaRDD<WriteStatus>> writeStatusRDDStream = allOf(clusteringPlan.getInputGroups().stream()
         .map(inputGroup -> runClusteringForGroupAsync(inputGroup,
             clusteringPlan.getStrategy().getStrategyParams(),
             Option.ofNullable(clusteringPlan.getPreserveHoodieMetadata()).orElse(false),
-            instantTime))
-        .map(CompletableFuture::join);
-
+            instantTime)).collect(Collectors.toList())).join().stream();
```

Comment:
Can you please re-format this snippet to stack up callers so that it's easy to attribute what method is invoked on each expression? 

Like following: 

```
allOf(
  clusteringPlan.getInputGroups().stream()
    .map(inputGroup -> runClusteringForGroupAsync(inputGroup,
            clusteringPlan.getStrategy().getStrategyParams(),
            Option.ofNullable(clusteringPlan.getPreserveHoodieMetadata()).orElse(false),
            instantTime)
    )
    .collect(Collectors.toList()))
    .join()
    .stream()
```

It's very hard to understand what is going on there right now